### PR TITLE
Add support for `TypedArray` and `ArrayBuffer`

### LINKED ISSUE #6: ArrayBuffers or Uint8Array support

### PR DESCRIPTION
**Pull request by maccelerated**
Fixes #6

### CODE REVIEW

**Comment by maccelerated** [comment]
I'm assuming the consumer of a stream will handle Endianness changes.

File: readme.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ intoStream('unicorn').pipe(process.stdout);
 
 ### intoStream(input)
 
-Type: `Buffer` `string` `Iterable<Buffer|string>` `Promise`<br>
+Type: `Buffer` `TypedArray` `string` `Iterable<Buffer|string>` `Promise`<br>
```
Comment (sindresorhus) [inline readme.md]: This should mention `ArrayBuffer` too, as TypedArray doesn't include ArrayBuffer. ArrayBuffer is the backing-store of a TypedArray.

File: index.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ module.exports = input => {
 
 	function prepare(value) {
 		input = value;
+		// Convert other buffer types to Buffer
+		input = input instanceof ArrayBuffer ? Buffer.from(input) : input;
+		input = !Buffer.isBuffer(input) && ArrayBuffer.isView(input) ? Buffer.from(input) : input;
```
Comment (sindresorhus) [inline index.js]: I would write this as:

```js
if (
	input instanceof ArrayBuffer ||
	(!Buffer.isBuffer(input) && ArrayBuffer.isView(input))
) {
	input = Buffer.from(input);
}
```

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**